### DISCUSSION TITLE
How to specify device?

### DISCUSSION THREAD

**Discussion by GuiltyImp** [rupeshs/fastsdcpu#119] [Q&A] [answered]
I would like to know how to specify which device is used when running this application. 

Is there a line of code I can add to the start batch file that lets me specify if I would like to use CPU, GPU.0, GPU.1, or GPU.*? 

Right now it seems to be defaulting to "cpu".

**Comment by rupeshs** [comment, marked as answer]
@GuiltyImp  For that you need to set the DEVICE environment variable.
If you are on Windows `set DEVICE=GPU.0`
For Linux `export DEVICE=GPU.0`

**Reply by GuiltyImp** [reply]
Thank you so much for the quick response. I haven't tested the app fully yet, but this is displaying device changed in terminal which is great. Thank you again!
